### ISSUE TITLE
Trailing commas and multiple lines for collections and enums

### ISSUE THREAD

**Issue by ZevEisenberg** [Raizlabs/Raizlabs-Cocoa-Style#40]
@adamhrz and I have been using this pattern, and we really like it. It takes advantage of the fact that you can leave a trailing comma at the end of Objective-C collection literals and enums, and the compiler doesn't complain. The advantage is that, if you add or delete an item anywhere in the collection, it dirties only that line. It also makes collection literals more readable, since you always expect them to look the same:

**Not** this:

``` objc
typedef NS_ENUM(NSUInteger, RZThingy) {
    RZThingyUndefined = 0,
    RZThingyStuff,
    RZThingyAnother
}
NSArray *foo = @[thing1, thing2, thing3];
NSDictionary *bar = @{key1: value1, key2: value2};
```

This:

``` objc
typedef NS_ENUM(NSUInteger, RZThingy) {
    RZThingyUndefined = 0,
    RZThingyStuff,
    RZThingyAnother, // ← note trailing comma
}

NSArray *foo = @[
                 thing1,
                 thing2,
                 thing3, // ← note trailing comma
                 ];

NSDictionary *bar = @{
                      key1: value1,
                      key2: value2, // ← note trailing comma
                      };
```

Question for @mgorbach @ateliercw @KingOfBrian: can we enforce this with clang-format/Faux Pas/whatever?

**Comment by KingOfBrian** [comment]
:+1: on the style!

There's no way to enforce with clang-format, not sure about Faux Pas.  There is something called 'clang-tidy' that may be able to enforce it with some work.

**Comment by ZevEisenberg** [comment]
SwiftLint is handling this for us in Swift. Not worth spending time trying to figure out how to enforce it in Obj-C.
